contracts/fee_currencies.go

```diff
@@ -23,30 +22,32 @@ const (
 	IntrinsicGasForAlternativeFeeCurrency uint64 = 50 * Thousand
 )
 
-var (
-	tmpAddress = common.HexToAddress("0xce106a5")
-)
-
 // Debits transaction fees from the transaction sender and stores them in the temporary address
 func DebitFees(evm *vm.EVM, feeCurrency *common.Address, address common.Address, amount *big.Int) error {
 	if amount.Cmp(big.NewInt(0)) == 0 {
 		return nil
 	}
-	abi, err := abigen.FeeCurrencyMetaData.GetAbi()
+	tokenAbi, err := abigen.FeeCurrencyMetaData.GetAbi()
 	if err != nil {
 		return err
 	}
-	// Solidity: function transfer(address to, uint256 amount) returns(bool)
-	input, err := abi.Pack("transfer", tmpAddress, amount)
+	// Solidity: function debitGasFees(address from, uint256 value)
+	input, err := tokenAbi.Pack("debitGasFees", address, amount)
 	if err != nil {
 		return err
```

Comment:
for `credit` you enrich this error a bit, I think it would make sense here as well.